lib/workers/repository/process/lookup/index.ts

```diff
@@ -329,7 +330,47 @@ export async function lookupUpdates(
       const sortedReleases = bucketReleases.sort((r1, r2) =>
         versioning.sortVersions(r1.version, r2.version)
       );
-      const bucketRelease = sortedReleases.pop();
+      let bucketRelease: Release;
+      let skippedOverVersions: string[];
+      let pendingVersions: string[];
+      // TODO: add this to config definitions
+      if (config.eagerStatusChecks) {
+        // If some releases satisfy checks and some don't, then suppress the non-satisfying
+        const pendingChecks: string[] = [];
+        if (config.stabilityDays) {
+          const ONE_DAY = 24 * 60 * 60 * 1000;
+          // Check each release for stabilityDays
+          for (const release of sortedReleases) {
+            // stabilityDays is only valid if a releaseTimestamp is present
+            if (release.releaseTimestamp) {
+              const daysElapsed = Math.floor(
+                (new Date().getTime() -
+                  new Date(release.releaseTimestamp).getTime()) /
+                  ONE_DAY
+              );
+              if (daysElapsed < config.stabilityDays) {
+                pendingChecks.push(release.version);
+              }
+            }
```

Comment:
This should be done with `luxon` as it's much easier and cleaner. can send a sample later

Reply:
https://github.com/renovatebot/renovate/blob/6fa37294cfb103f5b7a3c8e8b186303bfa4842ce/lib/workers/branch/index.ts#L2
https://github.com/renovatebot/renovate/blob/6fa37294cfb103f5b7a3c8e8b186303bfa4842ce/lib/workers/branch/index.ts#L456-L457